app/domain/authentication/authn/authenticator.rb

```diff
@@ -5,34 +5,34 @@
 module Authentication
   module Authn
 
-    class Authenticator < ::Dry::Struct
+    class Authenticator
 
       def self.requires_env_arg?
         false
       end
 
-      # optional
-      #
-      attribute :role_cls, ::Types::Any.default{ ::Authentication::MemoizedRole }
-      attribute :credentials_cls, ::Types::Any.default { ::Credentials }
+      def initialize(role_cls: Role, credentials_cls: ::Credentials)
+        @role_cls = role_cls
+        @credentials_cls = credentials_cls
+      end
 
       # Authenticates a Conjur using their username and password
       def login(input)
-        credentials = credentials(input)
-        success = credentials&.authenticate(input.password)
+        creds = credentials(input)
+        success = creds&.authenticate(input.password)
 
-        success ? credentials.api_key : nil
+        success ? creds.api_key : nil
```

Comment:
So to *really* get rid of nil checks we'd need to be using sum types, but that's a bigger change we won't be doing (and don't do anywhere).   That said, I think we can make reek happy (and make the code more explicit) with this change:
```ruby
      def login(input)
        creds = credentials(input)
        return nil unless creds

        success = creds.authenticate(input.password)
        success ? creds.api_key : nil
      end
```